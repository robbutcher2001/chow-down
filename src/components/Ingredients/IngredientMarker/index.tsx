import React from 'react';

import styled from 'styled-components';

interface IngredientMarkerProps {
  mark: string
};

const IngredientMarker = styled.li`
  font-size: ${props =>
    props.theme.typography.fontSize.xxlarge
  };
  border-left-style: solid;
  border-left-width: 0.5rem;
  border-left-color: #ca4a6c;
  margin: 3rem 0.5rem 0.5rem 0.5rem;
  padding: 1rem;
  flex: 0 1 100%;
`

export default (props: IngredientMarkerProps) => (
  <IngredientMarker>
    {props.mark}
  </IngredientMarker>
);