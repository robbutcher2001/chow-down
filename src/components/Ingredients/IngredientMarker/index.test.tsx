import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import IngredientMarker from '.';

test('IngredientMarker basic snapshot render', () => {
  const ingredientMarker = renderer.create(
    <IngredientMarker mark={'A_MARKER'} />
  );

  expect(ingredientMarker.toJSON()).toMatchSnapshot();
});

test('IngredientMarker correct tag content assertion', () => {
  const { getByText } = render(
    <IngredientMarker mark={'A_MARKER'} />
  );

  expect(getByText(/A_MARKER/).textContent).toEqual('A_MARKER');
});