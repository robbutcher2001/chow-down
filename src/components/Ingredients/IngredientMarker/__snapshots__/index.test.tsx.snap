// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientMarker basic snapshot render 1`] = `
<li
  className="sc-bdVaJa bnvtbd"
>
  A_MARKER
</li>
`;
